docs/advanced/advanced_running_stateful_filesystem.md

```diff
@@ -0,0 +1,106 @@
+## Running a stateful container with file system configurations
+
+1. Follow the instructions for running a stateful container, as detailed in steps 1 through 4 of [Sample configurations for running a stateful container](../content/using/csi_ug_using_sample.md).
+
+2. Create a StatefulSet.
+
+    <pre>
+    $> kubectl create -f demo-statefulset-file-system.yaml
+    statefulset.apps/demo-statefulset-file-system created
+        
+    $> cat demo-statefulset-file-system.yaml
+    kind: StatefulSet
+    apiVersion: apps/v1
+    metadata:
+      name: demo-statefulset-file-system
+    spec:
+      selector:
+        matchLabels:
+          app: demo-statefulset
+      serviceName: demo-statefulset
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: demo-statefulset
+        spec:
+          containers:
+          - name: demo-container
+            image: registry.access.redhat.com/ubi8/ubi:latest
+            command: [ "/bin/sh", "-c", "--" ]
+            args: [ "while true; do sleep 30; done;" ]
+            <b>volumeMounts:
+              - name: demo-volume-file-system
+                mountPath: "data"</b>
+          volumes:
+          - name: demo-volume-file-system
+            persistentVolumeClaim:
+              claimName: demo-pvc-file-system
+      #      nodeSelector:
+      #        kubernetes.io/hostname: HOSTNAME
+3. Check the newly created pod.
+
+    Display the newly created pod (make sure the pod status is _Running_).
+
+    <pre>
+    $> kubectl get pod demo-statefulset-file-system-0
+    NAME                 READY   STATUS    RESTARTS   AGE
+    demo-statefulset-file-system-0   1/1     Running   0          43s
+
+4. Write data to the persistent volume of the pod.
+
+    The PV should be mounted inside the pod at `/data`.
+
+    <pre>
+    $> kubectl exec demo-statefulset-file-system-0 -- touch /data/FILE
+    $> kubectl exec demo-statefulset-file-system-0 -- ls /data/FILE
+    /data/FILE
+
+5. Delete StatefulSet and then recreate, in order to validate data (/data/FILE) remains in the persistent volume.
+
+    1. Delete the StatefulSet.
+
+        <pre>
+        $> kubectl delete statefulset/demo-statefulset-file-system
+        statefulset/demo-statefulset-file-system deleted
+
+    2. Wait until the pod is deleted. Once deleted, the `"demo-statefulset-file-system" not found` is returned.
+
+        <pre>
+        $> kubectl get statefulset/demo-statefulset-file-system
+        Error from server (NotFound): statefulsets.apps <StatefulSet name> not found
+
+    3. Verify that the multipath was deleted and that the PV mountpoint no longer exists by establishing an SSH connection and logging into the worker node.
+          
+        <pre>
+        $> ssh root@k8s-node1
+            
+        $>[k8s-node1] df | egrep pvc
```

Comment:
@roysahar could this show non-ibm-block-csi content?
if so, could we avoid it?

Reply:
@roysahar  @oriyarde  Please take into account that this is all specifically _advanced_ content. Thanks

Reply:
advanced or not, IMO we should avoid irrelevant outputs wherever we can

Reply:
@oriyarde I am not sure how other csi drivers implement the final mount path, we are using in the mount point with the pv name that is always `pvcXXXX`. If the mount path on other csi drivers also use the pv name with `pvc` in it the `egrep` output will include them also.

Reply:
>I am not sure how other csi drivers implement the final mount path, we are using in the mount point with the pv name that is always `pvcXXXX`

AFAIK csi drivers are not in control of the paths, they get it from the CO (kubelet, in our case).


>If the mount path on other csi drivers also use the pv name with pvc in it the egrep output will include them also.

hence [my question](https://github.com/IBM/ibm-block-csi-driver/pull/387#discussion_r757909965): could we avoid showing non-ibm-block-csi content?

Reply:
@oriyarde I am working on find the right `jsonpath` command to get all `block.csi.ibm` provisioner PVCs

Reply:
you mean:
`kubectl get pods <pod-name> -n <namespace> -o jsonpath='{.metadata.uid}'`
and then:
`ls -l /var/lib/kubelet/pods/<pod-uid>/volumes/kubernetes.io~csi/`
?

Reply:
No, I mean to get all PVCs that their provisioner is our driver. I will put example soon.

Reply:
@oriyarde here is an example how to get the provisioner of all PVCs:
`kubectl get pvc -o jsonpath='{.items[*].metadata.annotations.volume\.beta\.kubernetes\.io\/storage-provisioner}'`
`block.csi.ibm.com` 
We can use this in oneliner to get all PVCs that their provisioner is `block.csi.ibm.com` .
What do you think?

Reply:
LGTM